hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastViewExpander.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.validate;
+
+import com.google.common.collect.ImmutableList;
+import com.hazelcast.jet.sql.impl.parse.QueryParseResult;
+import com.hazelcast.jet.sql.impl.parse.QueryParser;
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.schema.ViewResolver;
+import com.hazelcast.sql.impl.schema.view.View;
+import org.apache.calcite.sql.SqlAsOperator;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlJoin;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSelect;
+
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Hazelcast view expander helper class.
+ */
+public class HazelcastViewExpander {
+    private final HazelcastSqlValidator validator;
+    private final Set<View> visitedViews = new HashSet<>();
+    private final QueryParser parser;
+
+    public HazelcastViewExpander(HazelcastSqlValidator validator) {
+        this.validator = validator;
+        this.parser = new QueryParser(validator);
+    }
+
+    /**
+     * Tries to detect and expand view in given {@link SqlSelect} argument.
+     * View expanding happens on AST IR level.
+     */
+    @SuppressWarnings("CheckStyle")
+    public void expandView(SqlSelect selectCall) {
```

Comment:
We miss handling for view references in the SELECT clause, in the WHERE clause (in EXISTS expr.), and most importantly, as an argument to table functions, e.g.:
```sql
select *
from table(TUMBLE(table(my_view), ...)
```